File: frontend/src/types/analysis.ts
export interface TechnicalIndicatorValues {
  sma_20: number | null;
  sma_50: number | null;
  rsi_14: number | null;
  macd: number | null;
  macd_signal: number | null;
  volatility_20: number | null;
}

export interface TechnicalAnalysisResponse {
  symbol: string;
  date: string;
  technical_score: number;
  technical_signal: "BULLISH" | "NEUTRAL" | "BEARISH";
  indicators: TechnicalIndicatorValues;
  reasons: string[];
}

File: backend/app/schemas/analysis.py
from datetime import date

from pydantic import BaseModel


class TechnicalIndicatorValues(BaseModel):
    sma_20: float | None
    sma_50: float | None
    rsi_14: float | None
    macd: float | None
    macd_signal: float | None
    volatility_20: float | None


class TechnicalAnalysisResponse(BaseModel):
    symbol: str
    date: date
    technical_score: float
    technical_signal: str
    indicators: TechnicalIndicatorValues
    reasons: list[str]


class TechnicalRankingItem(BaseModel):
    rank: int
    symbol: str
    date: date
    technical_score: float
    technical_signal: str


class TechnicalRankingResponse(BaseModel):
    as_of_date: date
    rankings: list[TechnicalRankingItem]
